Cassette/CASFileObjectQueue.m

```diff
@@ -50,12 +54,28 @@ - (void)add:(id)data {
     [self.queueFile add:[NSKeyedArchiver archivedDataWithRootObject:data]];
 }
 
+- (void)addArchived:(NSData *)data {
+    [self.queueFile add:data];
+}
+
+- (NSData *)peekArchivedData {
+    NSArray<id> *elements = [self peek:1 archived:YES];
+    if (elements.count > 0) {
+        return elements[0];
+    }
+    return nil;
+}
+
 - (NSArray<id> *)peek:(NSUInteger)amount {
+    return [self peek:amount archived:NO];
+}
+
+- (NSArray<id> *)peek:(NSUInteger)amount archived:(BOOL)archived {
     NSArray<NSData *> *elements = [self.queueFile peek:amount];
     NSMutableArray<id> *coercedElements = [[NSMutableArray alloc] init];
     for (NSUInteger i = 0; i < elements.count; i++) {
         NSData *element = elements[i];
-        id coercedElement = [NSKeyedUnarchiver unarchiveObjectWithData:element];
+        id coercedElement = archived ? element : [NSKeyedUnarchiver unarchiveObjectWithData:element];
```

Comment:
I see what you're doing, and how it may result in better performance. I suggest either using CASQueueFile directly, or building a a CASFileDataQueue rather than changing the API here.

I imagine users want to either store NSData or Objects, not both. If your use case desires both, please extend that in your own application

Reply:
I could certainly do this as a subclass for my own usecase, however I not convinced mine is an entirely unique usecase here. I can imagine people would want to transfer data from one queue to another, or from one storage medium to another. This removed the overhead of having to do extra archiving and unarchiving for those cases. If the objects are quite large that overhead and be quite a lot.